iothub/device/devdoc/API Design.md

```diff
@@ -187,95 +402,141 @@ public string AckDescription { get; set; }
+    }
+
+    /// <summary>
+    ///
+    /// </summary>
+    /// <param name="objectToSendWithConvention"></param>
+    /// <returns></returns>
+    public virtual byte[] GetObjectBytes(object objectToSendWithConvention)
+    {
+        return EncodeStringToByteArray(SerializeToString(objectToSendWithConvention));
+    }
+}
 ```
+</details>
 
 ### Commands
 
 ```csharp
 /// <summary>
-/// Set command callback handler.
-/// </summary>
-/// <param name="commandName">The name of the command this handler will be used for.</param>
-/// <param name="commandCallback">The callback for this command.</param>
-/// <param name="componentName">The component name this command belongs to.</param>
-/// <param name="userContext">Generic parameter to be interpreted by the client code.</param>
-/// <param name="cancellationToken"></param>
-/// <remarks>
-/// The .NET SDK has a built in dispatcher that handles per command name routing. The SDK will first attempt to find the command by name. If it is found it will execute the callback for that specific command. If there is no entry found in the dispatcher the SDK will fall back to the global command handler <see cref="SetCommandCallbackHandlerAsync(Func{CommandRequest, object, Task{CommandResponse}}, object, CancellationToken)"/>.
-/// </remarks>
-public Task SetCommandCallbackHandlerAsync(string commandName, Func<CommandRequest, object, Task<CommandResponse>> commandCallback, string componentName = default, object userContext = default, CancellationToken cancellationToken = default);
-
-/// <summary>
-/// Set the global command callback handler. This handler will be called when no named handler was found for the command.
+/// Set the global command callback handler.
 /// </summary>
-/// <param name="commandCallback">A method implementation that will handle the incoming command.</param>
+/// <param name="callback">A method implementation that will handle the incoming command.</param>
 /// <param name="userContext">Generic parameter to be interpreted by the client code.</param>
 /// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
-/// <remarks>
-/// The global command handler will be the fallback handler in the event there is nothing specified in the dispatcher. See the remarks in <see cref="SetCommandCallbackHandlerAsync(Func{CommandRequest, object, Task{CommandResponse}}, object, CancellationToken)"/> for more information.
-/// </remarks>
-public Task SetCommandCallbackHandlerAsync(Func<CommandRequest, object, Task<CommandResponse>> commandCallback, object userContext = default, CancellationToken cancellationToken = default);
+public Task SubscribeToCommandsAsync(Func<CommandRequest, object, Task<CommandResponse>> callback, object userContext, CancellationToken cancellationToken = default)
 ```
 
 <details>
 <summary>CommandRequest</summary>
 
 ```csharp
-public sealed class CommandRequest : MethodRequest
+public sealed class CommandRequest
```

Comment:
no longer inherting from MothodRequest, since MethodRequest has other properties defined (response timeout, connection timeout) which aren't relevant for a device-application, and reimplementing it seems better than inheriting and hiding (oop concept-wise)